### PR TITLE
Add text support

### PR DESCRIPTION
**Pull request by foxhatleo**
This PR adds text support. Level editor will be updated soon to support this.

### CODE REVIEW

File: core/src/com/fallenflame/game/TextController.java [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+import org.w3c.dom.css.Rect;
+
+import java.util.*;
+
+public class TextController {
+    private Map<Rectangle, String> texts;
+    private String nextMessage = null;
+    private BitmapFont displayFont;
+    private int screenWidth;
+    private int screenHeight;
```
Comment (BryanGraeser) [inline core/src/com/fallenflame/game/TextController.java]: It doesn't look like screenHeight is used

File: core/src/com/fallenflame/game/TextController.java [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+    }
+
+    public void update(PlayerModel player) {
+        if (texts == null) return;
+        for (Map.Entry<Rectangle, String> ele : texts.entrySet()) {
+            if (ele.getKey().contains(player.getX(), player.getY())) {
+                nextMessage = ele.getValue();
+                return;
+            }
+        }
```
Comment (BryanGraeser) [inline core/src/com/fallenflame/game/TextController.java]: So if I'm reading this right, do the player X and Y coordinates have to be exactly the value stored in order to transition the message? Having it required to be an exact match may be challenging for the player to actually trigger

File: core/src/com/fallenflame/game/TextController.java [hunk 3]
```diff
@@ -0,0 +1,87 @@
+package com.fallenflame.game;
+
+import com.badlogic.gdx.Gdx;
+import com.badlogic.gdx.graphics.Color;
+import com.badlogic.gdx.graphics.g2d.BitmapFont;
+import com.badlogic.gdx.graphics.g2d.GlyphLayout;
+import com.badlogic.gdx.math.Rectangle;
+import com.badlogic.gdx.math.Vector2;
```
Comment (BryanGraeser) [inline core/src/com/fallenflame/game/TextController.java]: Doesn't look like this is used

File: core/src/com/fallenflame/game/TextController.java [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+import com.badlogic.gdx.utils.JsonValue;
+import com.fallenflame.game.util.JsonAssetManager;
+import org.w3c.dom.css.Rect;
```
Comment (BryanGraeser) [inline core/src/com/fallenflame/game/TextController.java]: Or this

**Review by BryanGraeser — COMMENTED** [review]
Few minor comments and one question: how precise does the player coordinate have to be to trigger the next message?

Also do you have a json with text as an example I can test this with?

File: core/src/com/fallenflame/game/TextController.java (hunk 3 above)
Comment (foxhatleo) [reply core/src/com/fallenflame/game/TextController.java]: I will optimise the imports.

File: core/src/com/fallenflame/game/TextController.java (hunk 2 above)
Comment (foxhatleo) [reply core/src/com/fallenflame/game/TextController.java]: It's not an exact value. The key is a rectangle so it is a box.

**Review by BryanGraeser — APPROVED** [review]
gtg

**Pull request merged**